### PR TITLE
Added Metamask verification, pending testing with frontend updates

### PR DESCRIPTION
**Pull request by rfun**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -42,211 +49,230 @@ let transporter = nodemailer.createTransport({
...
 
 
 //////////////////
 // Static Files //
 //////////////////
 
-app.get('/', function (req, res) {
-  res.sendFile(path.join(__dirname + '/build/index.html'));
+app.get('/', function(req, res) {
+    res.sendFile(path.join(__dirname + '/build/index.html'));
```
Comment (joshuakcockrell) [inline app.js]: Did you change all the 2 space indents into 4 spaces? Is that what all this is?

File: app.js [hunk 2]
```diff
@@ -320,24 +348,24 @@ app.get('/battles/history', (req, res) => {
...
-      ethAddress: req.body.ethAddress,
-      userEthAddress: user.ethAddress,
-      color: req.body.color,
-      experience: req.body.experience,
-      level: req.body.level
-    }).then(newShape => { 
-      return res.status(200).end();
-    }).catch(err => {
-      console.log(err);
+    console.log('');
```
Comment (joshuakcockrell) [inline app.js]: A bunch of chunks are modified. Did anything change functionality wise here?

**Comment by joshuakcockrell** [comment]
The authentication stuff looks good! I'm okay to merge this after @rfun clarifies if he changed anything other than the visual look of the code from 2 to 4 spaces.

File: app.js (hunk 1 above)
Comment (rfun) [reply app.js]: Sorry yeah all of these are just spacing changes. Would you like me to restore those?

Comment (joshuakcockrell) [reply app.js]: No I think it's fine. @trevorlyman and I are the only other ones that touch this app.js file so it shouldn't cause too many merge conflicts later.

**Pull request merged**